src/main/antlr4/liquid/parser/v4/LiquidLexer.g4

```diff
@@ -188,7 +188,10 @@ mode IN_TAG;
 mode IN_TAG_ID;
   WS2 : WhitespaceChar+ -> channel(HIDDEN);
 
-  CommentInTagId : ( '#' ( {!linebreakOrTagEndAhead()}? . )* [ \t\r\n]* )+ -> channel(HIDDEN);
+  // When we encounter a comment in this "IN_TAG_ID" mode, there is no need to try to create more tokens
+  // since that will always result in an `InvalidEndTag` token. After all, a "#" comment consumes all
+  // characters until it "sees" an end-tag. Just pop out in that case.
+  CommentInTagId : ( '#' ( {!linebreakOrTagEndAhead()}? . )* [ \t\r\n]* )+ -> channel(HIDDEN), popMode;
```

Comment:
where's this function definition or implementation?

Reply:
It was part of a previous PR. It's in the lexer grammar: https://github.com/bkiers/Liqp/blob/master/src/main/antlr4/liquid/parser/v4/LiquidLexer.g4#L68-L73